src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/reduce/reduce_kernel_b_fs_yx_fsv16.cpp

```diff
@@ -211,7 +211,14 @@ JitConstants ReduceKernel_b_fs_yx_fsv16::GetJitConstants(const reduce_params& pa
 }
 
 KernelsData ReduceKernel_b_fs_yx_fsv16::GetKernelsData(const Params& params, const optional_params& options) const {
-    return GetCommonKernelsData(params, options);
+    KernelsData kds = GetCommonKernelsData(params, options);
+    const reduce_params& orgParams = static_cast<const reduce_params&>(params);
+
+    if (orgParams.inputs[0].Feature().v % 16 != 0) {
+        kds[0].can_reuse_memory = false;
```

Comment:
The solution looks more like a work-around. Technically, we can reuse such memory buffers, we just need to ensure that we have one of the following:
- paddings over feature dimension contains zeros. i.e. each kernel should guarantee this it doesn't write anything to the pad area
- kernels sensitive to values in the pad area mustn't read the data from pads.

Since the first option is hard to maintain, I think we need to focus on the second one.

Reply:
Agreed that this is more like workaround. The ticket issue(80787) also happened after https://github.com/openvinotoolkit/openvino/pull/9359. Rollback affects to performance a lot. So we introduced this PR.